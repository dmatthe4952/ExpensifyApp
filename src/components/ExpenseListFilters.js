import React from 'react';
import { connect } from 'react-redux';
import {setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate} from '../actions/filters';
import {DateRangePicker} from 'react-dates';

export class ExpenseListFilters extends React.Component {
    state={
        calendarFocused: null
    }

    onDatesChange = ({startDate, endDate})=>{
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)
    }
    onFocusChange = (calendarFocused)=>{
        this.setState(() =>({calendarFocused}))
    }
    onTextFilterChange = (e) => {
        this.props.setTextFilter(e.target.value);
    }
    onSortByFilterChange = (e) => {
        if (e.target.value === 'date'){
            this.props.sortByDate();
        } else  if (e.target.value === 'amount'){
            this.props.sortByAmount();
        }
    }
    render(){
        return (
            <div>
                <span>
                Filter:
                <input
                    type='text'
                    value={this.props.filters.text}
                    onChange={this.onTextFilterChange}/>
                </span>
                <span>
                SortBy:
                <select 
                    value={this.props.filters.sortBy}
                    onChange={this.onSortByFilterChange}>
                    <option value='date' >Date</option>
                    <option value='amount' >Amount</option>
                </select>
                </span>
                <span>
                DateRange:
                <DateRangePicker 
                    startDate={this.props.filters.startDate}
                    endDate={this.props.filters.endDate}
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    showClearDates={true}
                    numberOfMonths={1}
                    isOutsideRange={()=>{false}}
                    />
                </span>
            </div>
        )
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        setStartDate: (startDate)=>(dispatch(setStartDate(startDate))),
        setEndDate: (endDate)=>(dispatch(setEndDate(endDate))),
        setTextFilter: (text)=>(dispatch(setTextFilter(text))),
        sortByDate: ()=>(dispatch(sortByDate())),
        sortByAmount: ()=>(dispatch(sortByAmount()))
    }
}

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
};

export default connect(mapStateToProps,mapDispatchToProps)(ExpenseListFilters);