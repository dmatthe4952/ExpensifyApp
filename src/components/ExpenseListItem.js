import React from 'react';
import { connect } from 'react-redux';
import moment from 'moment';
import {startRemoveExpense} from '../actions/expenses';
import {Link} from 'react-router-dom';
import numeral from 'numeral';


export const ExpenseListItem = ({dispatch, description, id, amount, createdAt, note}) => {
    return (
        <tr>
            <td><Link to={`/edit/${id}`}>{description}</Link></td>
            <td>{numeral(amount / 100).format('$0,0.00')}</td>
            <td>{moment(createdAt).format("MM/DD/YYYY")}</td>
            <td>{note}</td>
            <td><button onClick={(e)=>{
                dispatch(startRemoveExpense({id}));
            }}>
            Remove</button></td>
        </tr>
    );
};

const mapStateToProps = (state) => {
    return {
        expenses: state.expenses
    };
};

export default connect()(ExpenseListItem);