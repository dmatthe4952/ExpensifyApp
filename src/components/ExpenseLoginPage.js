import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const ExpenseLoginPage  = ({startLogin}) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__box__title" >Track My Spending</h1>
            <p>Keep your spending under control by tracking your spending habits.</p>
            <button onClick={startLogin} id="login-button" >Login with Google</button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps) (ExpenseLoginPage);