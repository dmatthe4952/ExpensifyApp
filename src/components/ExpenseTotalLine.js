import React from 'react';
import { connect } from 'react-redux';
import expensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral';

export const ExpenseTotalLine = (props) => (
    <div>
    {
        props.expenses.length === 0 ?
        (<p/>) : 
        (<p>You are viewing {props.expenses.length} expenses,
            totaling {numeral(expensesTotal(props.expenses)/100).format('$0,0.00')}.</p>)
    }
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseTotalLine);