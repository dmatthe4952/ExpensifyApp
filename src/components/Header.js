import React from 'react';
import {NavLink} from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({startLogout}) => (
    <header className="header">
        <h1>Track My Spending</h1>
        <NavLink to="/dashboard" activeClassName="is-active" exact={true}> Home </NavLink>
        <NavLink to = "/create" activeClassName="is-active"> Add an Expense </NavLink>
        <NavLink to = '/help' activeClassName='is-active'> Help </NavLink>
        <button id="logout" onClick={startLogout}  >LogOut</button>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
        startLogout: () =>  dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);