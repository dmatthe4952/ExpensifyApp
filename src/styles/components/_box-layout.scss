

.box-layout {
    align-items: center;
    background: url('/images/dollar-2584746_1920.jpg');
    background-size: cover;
    display: flex;
    height: 100vh;
    justify-content: center;
    width: 100vw;
}

.box-layout__box {
    background: fade-out(white, .15);
    border-radius: 4px;
    text-align: center;
    padding: 3.2rem, 4.6rem;
    width: 25rem;
}

.box-layout__box__title {
    margin-top: 0;
    background-color: $dark-grey;
    padding-top: 3px;
    color: #dfdfdf;
}