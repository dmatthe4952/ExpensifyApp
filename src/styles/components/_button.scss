.button {
    cursor: pointer;
    background-color: rgb(41, 106, 128);
    border-radius: 3px;
    text-decoration: none;
    color: white;
    display: inline-block;
    font-size: $l-size-font;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    border: none;
    border-radius: 5px;
}

.button--link {
    background: none;
}

.button--secondary {
    background: #888;
}