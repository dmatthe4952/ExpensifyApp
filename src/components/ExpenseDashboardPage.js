import React from 'react';
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters';
import ExpenseTotalLine from './ExpenseTotalLine';

const ExpenseDashboardPage = () => (
    <div className="content-container">
        <ExpenseTotalLine />
        <ExpenseListFilters />        
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;