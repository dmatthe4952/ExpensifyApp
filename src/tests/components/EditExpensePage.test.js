import React from 'react';
import { shallow }  from 'enzyme';
import { EditExpensePage } from '../../components/EditExpensePage';
import expenses from '../fixtures/expenses';

let startEditExpense, startRemoveExpense, wrapper, history
beforeEach(()=>{
    startEditExpense = jest.fn();
    startRemoveExpense = jest.fn();
    history = {push: jest.fn()};
    wrapper = shallow(<EditExpensePage
         startEditExpense={startEditExpense}
         startRemoveExpense={startRemoveExpense}
         history={history}
         expense={expenses[1]}
         />); 
})

test('should render EditExpensePage correctly', ()=> {
    expect(wrapper).toMatchSnapshot();
});

test('should handle onSubmit if called', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1])
    expect(startEditExpense).toHaveBeenLastCalledWith(expenses[1].id,expenses[1]);
    expect(history.push).toHaveBeenLastCalledWith('/');
});

test('should handle onClick when remove pressed',()=>{
    wrapper.find('button').prop('onClick')({id:expenses[1].id})
    expect(startRemoveExpense).toHaveBeenLastCalledWith({id:expenses[1].id})
    expect(history.push).toHaveBeenLastCalledWith('/')
});