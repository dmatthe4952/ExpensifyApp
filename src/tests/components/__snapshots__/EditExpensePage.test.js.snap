// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpensePage correctly 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 109500,
        "createdAt": -86400000,
        "description": "Rent",
        "id": 2,
        "note": "February Rent",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove Expense
  </button>
</div>
`;
